plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/TrinoJdbcCatalog.java

```diff
@@ -398,19 +446,48 @@ public void setViewPrincipal(ConnectorSession session, SchemaTableName schemaVie
     @Override
     public void dropView(ConnectorSession session, SchemaTableName schemaViewName)
     {
-        throw new TrinoException(NOT_SUPPORTED, "dropView is not supported for Iceberg JDBC catalogs");
+        jdbcCatalog.dropView(toIdentifier(schemaViewName));
     }
 
     @Override
     public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, Optional<String> namespace)
     {
-        return ImmutableMap.of();
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
+        for (Namespace ns : jdbcCatalog.listNamespaces()) {
+            for (TableIdentifier restView : jdbcCatalog.listViews(ns)) {
+                SchemaTableName schemaTableName = SchemaTableName.schemaTableName(restView.namespace().toString(), restView.name());
+                getView(session, schemaTableName).ifPresent(view -> views.put(schemaTableName, view));
```

Comment:
This logic may throw an exception if views with unsupported dialect exist in the JDBC catalog. 

Please take a look at REST catalog implementation & `TestIcebergTrinoRestCatalogConnectorSmokeTest#testUnsupportedViewDialect`:
https://github.com/trinodb/trino/blob/facddc80df514c0aa05341d74d0cb62004b63d8d/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java#L444-L453